pageserver/src/repository.rs

```diff
@@ -137,38 +137,26 @@ pub trait Timeline: Send + Sync {
     /// This method is used for marking dropped relations and truncated SLRU segments
     fn put_unlink(&self, tag: RelishTag, lsn: Lsn) -> Result<()>;
 
-    /// Remember the all WAL before the given LSN has been processed.
+    /// Track end of the latest digested WAL record.
     ///
-    /// The WAL receiver calls this after the put_* functions, to indicate that
-    /// all WAL before this point has been digested. Before that, if you call
-    /// GET on an earlier LSN, it will block.
-    fn advance_last_valid_lsn(&self, lsn: Lsn);
-    fn get_last_valid_lsn(&self) -> Lsn;
-    fn init_valid_lsn(&self, lsn: Lsn);
-
-    /// Like `advance_last_valid_lsn`, but this always points to the end of
-    /// a WAL record, not in the middle of one.
-    ///
-    /// This must be <= last valid LSN. This is tracked separately from last
-    /// valid LSN, so that the WAL receiver knows where to restart streaming.
+    /// Advance requires aligned LSN as an argument and would wake wait_lsn() callers.
+    /// Previous last record LSN is stored alongside the latest and can be read.
     fn advance_last_record_lsn(&self, lsn: Lsn);
     fn get_last_record_lsn(&self) -> Lsn;
-
-    // Like `advance_last_record_lsn`, but points to the start position of last record
     fn get_prev_record_lsn(&self) -> Lsn;
 
+    ///
+    /// Wait until the repository receives given LSN. Right now it is only used
+    /// for waiting before sending basebackup.
```

Comment:
As far as I can see we use it in get_page_at_lsn too, is this comment still relevant?